compiler/parser/parser.cpp

```diff
@@ -793,6 +801,57 @@ static void addModuleToDoneList(ModuleSymbol* module) {
 *                                                                             *
 ************************************** | *************************************/
 
+
+ModuleSymbol* parseIncludedSubmodule(const char* name) {
+  // save parser global variables to restore after parsing the submodule
+  BlockStmt*  s_yyblock = yyblock;
+  const char* s_yyfilename = yyfilename;
+  int         s_yystarlineno = yystartlineno;
+  ModTag      s_currentModuleType = currentModuleType;
+  const char* s_currentModuleName = currentModuleName;
+  int         s_chplLineno = chplLineno;
+  bool        s_chplParseString = chplParseString;
+  const char* s_chplParseStringMsg = chplParseStringMsg;
+  bool        s_currentFileNamedOnCommandLine = currentFileNamedOnCommandLine;
+
+  std::string curPath = yyfilename;
+
+  // compute the path of the file to include
+  size_t lastDot = curPath.rfind(".");
+  INT_ASSERT(lastDot < curPath.size());
+  std::string noDot = curPath.substr(0, lastDot);
+  std::string includeFile = noDot + "/" + name + ".chpl";
+
+  const char* modNameFromFile = filenameToModulename(curPath.c_str());
+  if (0 != strcmp(modNameFromFile, currentModuleName))
+    USR_FATAL("Cannot include module from module with different file name");
+
+  ModuleSymbol* ret = parseFile(astr(includeFile), currentModuleType,
+                                /* namedOnCommandLine */ false,
+                                /* include */ true);
+
+  ret->addFlag(FLAG_INCLUDED_MODULE);
+
+  // restore parser global variables
+  yyblock = s_yyblock;
+  yyfilename = s_yyfilename;
+  yystartlineno = s_yystarlineno;
+  currentModuleType = s_currentModuleType;
+  currentModuleName = s_currentModuleName;
+  chplLineno = s_chplLineno;
+  chplParseString = s_chplParseString;
+  chplParseStringMsg = s_chplParseStringMsg;
+  currentFileNamedOnCommandLine = s_currentFileNamedOnCommandLine;
+
+  return ret;
+}
+
+/************************************* | **************************************
+*                                                                             *
+*                                                                             *
+*                                                                             *
+************************************** | *************************************/
+
```

Comment:
Are these empty blocks just comments waiting to be written?  Is there value in adding a new one?

Reply:
AFAIK they divide regions of the file mostly about particular functions. Of course this is a style thing. I was just trying to fit into the style already existing in the file.